### DISCUSSION TITLE
Si_AutoKickNegativeKills - Autobans/Autokicks question

### DISCUSSION THREAD

**Discussion by Cornholio777** [data-bomb/Silica#22] [Q&A] [answered]
Right now when a player gets autokicked they are also put on the banned list.
Is this the intended behavior or can I change so it only kicks but players can rejoin? And perhaps if they get autokicked a second time then it will ban for example 10 minutes, and third time within the same 24h it can ban for 1h etc etc..
Perhaps parameters we can set ourselves in the config file? 

Thank you!

**Comment by data-bomb** [comment]
@Cornholio777 it was intentional for this to result in at least a temporary ban since it was a strong indicator of griefing (or, perhaps, the player being brand new to the game). The default Kick action (what happens when the host clicks the Kick button on a player) results in a temporary ban for the life of the server (but not permanent). However, the Ban plugin changes the default action to permaban. This could be a configurable option in both places to give the most flexibility when assigning a punishment here, so it would range from:
- Kick only (they can rejoin immediately)
- Temporary ban (for the life of the server)
- Permanent ban

**Comment by data-bomb** [comment]
Preference for the penalty (ban vs kick) added here https://github.com/data-bomb/Silica/commit/d46c7978be0676522e9ef7d8ddf2dae258970f29

**Reply by data-bomb** [reply]
Preferences for the penalty of the host's kick button added here (https://github.com/data-bomb/Silica/commit/bfb018357292eb5d7765ae27c8f6f94d8662f066)

**Reply by data-bomb** [reply, marked as answer]
For the following desired penalty on a negative kill threshold violation, do the following:
- Kick only (Grief_NegativeKills_Penalty_Ban = false)
- Temporary ban (Ban_HostKickButton_Permabans = false, Grief_NegativeKills_Penalty_Ban = true)
- Permanent ban (Ban_HostKickButton_Permabans = true, Grief_NegativeKills_Penalty_Ban = true)
